File: ios/RNBinancePay.m
#import <React/RCTBridgeModule.h>
#import <React/RCTUtils.h>

@interface RCT_EXTERN_MODULE(RNBinancePay, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(payCrypto:(NSString*)merchantId prepayId:(NSString*)prepayId timestamp:(int64_t)timestamp noncestr:(NSString*)noncestr certSn:(NSString*)certSn sign:(NSString*)sign redirectScheme:(NSString*)redirectScheme
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(transferCrypto:(NSString*)orderId type:(NSString*)type redirectScheme:(NSString*)redirectScheme
                  withResolver:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)
                  
@end

File: ios/RNBinancePay.swift
import BinancePaySDK

@objc(RNBinancePay)
class RNBinancePay: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }

    @objc(payCrypto:prepayId:timestamp:noncestr:certSn:sign:redirectScheme:withResolver:withRejecter:)
    func payCrypto(merchantId: String, prepayId: String, timestamp: Int64, noncestr: String, certSn: String, sign: String, redirectScheme: String, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
        
        let presentedViewController = RCTPresentedViewController();
        let view = presentedViewController?.view

        let parameters = OrderInitParameters(merchantId: merchantId, prepayId: prepayId, timestamp: timestamp, noncestr: noncestr, certSn: certSn, sign: sign, redirectScheme: redirectScheme)

        BinancePay.shared.pay(with: parameters, containerView: view!) { (result) in
           switch result {
           case .success:
                resolve("success")
           case .failure(let error):
                resolve("failure \(error)")
           }
        }
    }
    
    @objc(transferCrypto:type:redirectScheme:withResolver:withRejecter:)
    func transferCrypto(orderId: String, type: String, redirectScheme: String, resolve: @escaping RCTPromiseResolveBlock,reject: @escaping RCTPromiseRejectBlock) -> Void {
        
        let presentedViewController = RCTPresentedViewController();
        let view = presentedViewController?.view
        
        let parameters = C2CInitParameters(id: orderId, type: type, redirectScheme: redirectScheme)
        
        BinancePay.shared.pay(with: parameters, containerView: view!) { (result) in
            switch result {
            case .success:
                resolve("success")
            case .failure(let error):
                resolve("failure \(error)")
            }
        }
    }
}
